File: app/src/main/java/com/metahumanz/pacilread/ui/ActivityTransitionCompat.java
package com.metahumanz.pacilread.ui;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.os.Build;

public final class ActivityTransitionCompat {
    private ActivityTransitionCompat() {
    }

    @SuppressLint({"NewApi", "InlinedApi"})
    public static void overrideOpen(Activity activity, int enterAnim, int exitAnim) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
            activity.overrideActivityTransition(Activity.OVERRIDE_TRANSITION_OPEN, enterAnim, exitAnim);
        } else {
            activity.overridePendingTransition(enterAnim, exitAnim);
        }
    }

    @SuppressLint({"NewApi", "InlinedApi"})
    public static void overrideClose(Activity activity, int enterAnim, int exitAnim) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
            activity.overrideActivityTransition(Activity.OVERRIDE_TRANSITION_CLOSE, enterAnim, exitAnim);
        } else {
            activity.overridePendingTransition(enterAnim, exitAnim);
        }
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/ui/ActivityTransitionCompat.kt
package com.metahumanz.pacilread.ui

import android.annotation.SuppressLint
import android.app.Activity
import android.os.Build

object ActivityTransitionCompat {
    @JvmStatic
    @SuppressLint("NewApi", "InlinedApi")
    fun overrideOpen(activity: Activity, enterAnim: Int, exitAnim: Int) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
            activity.overrideActivityTransition(Activity.OVERRIDE_TRANSITION_OPEN, enterAnim, exitAnim)
        } else {
            @Suppress("DEPRECATION")
            activity.overridePendingTransition(enterAnim, exitAnim)
        }
    }

    @JvmStatic
    @SuppressLint("NewApi", "InlinedApi")
    fun overrideClose(activity: Activity, enterAnim: Int, exitAnim: Int) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
            activity.overrideActivityTransition(Activity.OVERRIDE_TRANSITION_CLOSE, enterAnim, exitAnim)
        } else {
            @Suppress("DEPRECATION")
            activity.overridePendingTransition(enterAnim, exitAnim)
        }
    }
}
